modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_pool.cc

```diff
@@ -0,0 +1,34 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+#include "modules/localization/msf/local_map/pyramid_map/pyramid_map_pool.h"
+
+namespace apollo {
+namespace localization {
+namespace msf {
+
+PyramidMapNodePool::PyramidMapNodePool(unsigned int pool_size,
+                                       unsigned int thread_size)
+    : BaseMapNodePool(pool_size, thread_size) {}
+
+PyramidMapNodePool::~PyramidMapNodePool() {}
+
+BaseMapNode* PyramidMapNodePool::AllocNewMapNode() {
+  return new PyramidMapNode();
```

Comment:
Why not use unique pointer or shared pointer to avoid manual deletion?

Reply:
we need to control the release of MapNode